app/scripts/extract_texts_script.py

```diff
@@ -1,34 +1,33 @@
 import json
 import os
+import sys
 import re           # for doing regex search
 from pathlib import Path
 
 
-def extract_texts():
-    srcs = [
-        'template',
-        'global',
-        'tour',
-        'data_list',
-    ]
-    results_all = []
+def main():
+    ConfigPath = Path(sys.argv[1])
+    with open(ConfigPath) as config_file:
+        data = json.load(config_file)
+
+        in_dir = Path(data['ComposeResult'])
+        out_dir = Path(data['ExtractResult'])
+
+    extract_texts(in_dir, out_dir)
+
+def extract_texts(in_dir, out_dir):
+    #get a list of all the files in our input directory
+    srcs = os.listdir(in_dir)
 
     for src in srcs:
-        results = process_file(src)
-        save_results(src, results)
+        results = process_file(src, in_dir)
+        save_results(src, results, out_dir)
         print_report(src, results)
-        results_all += results
-
-    # Result from all files
-    print(f"Processing 'all'")
-    print(f'Length: {len(results_all)}')
-    save_results('all', dedupe(results_all))
-    print_report('all', results_all)
 
-def process_file(src):
+def process_file(src, in_dir):
     print(f"Processing '{src}'")
 
-    input_file_path = Path(os.getcwd()) / 'input' / f'input_{src}.json'
+    input_file_path = in_dir / src
```

Comment:
It would simplify things if `process_file` just expects a full path to the source file - that way, you don't have to pass in `in_dir`.